packages/markers/src/browser/problem/problem-decorator.ts

```diff
@@ -127,10 +127,14 @@ export class ProblemDecorator implements TreeDecorator {
     }
 
     protected toDecorator(marker: Marker<Diagnostic>): TreeDecoration.Data {
+        // Determine if the given marker is for a resource of a container.
+        const isResource = Array.from(this.problemManager.getUris()).some(m => m === marker.uri);
```

Comment:
I'm wondering if there is a better way to do this.
It'd be easy if I had access to the `FileStat` and could determine if the marker's URI is a directory or not.

Reply:
In this case, do only files have a marker? Folders wouldn't match anything?

You could write `const isResource = new Set(this.problemManager.getUris()).has(marker.uri);` and document why this test filters file from folders.

I don't see a clean way to get information about the filestat synchronously. Best we could do is make a bunch of modification to get data about the current node being decorated, and see what field we could add to the node (folder etc...)